GoogleSignIn/Sources/GIDSignIn.m

```diff
@@ -443,11 +456,43 @@ + (GIDSignIn *)sharedInstance {
   static dispatch_once_t once;
   static GIDSignIn *sharedInstance;
   dispatch_once(&once, ^{
-    sharedInstance = [[self alloc] initPrivate];
+    GTMKeychainStore *keychainStore =
+        [[GTMKeychainStore alloc] initWithItemName:kGTMAppAuthKeychainName];
+    sharedInstance = [[self alloc] initWithKeychainStore:keychainStore];
   });
   return sharedInstance;
 }
 
+#pragma mark - Configuring and pre-warming
+
+#if TARGET_OS_IOS && !TARGET_OS_MACCATALYST
+- (void)configureWithCompletion:(nullable void (^)(NSError * _Nullable))completion {
+  @synchronized(self) {
+    // This check helps to avoid needing to set `_appCheck` in an initializer called in
+    // `+[GIDSignIn sharedInstance]`. Also defers to the fakes for `_appCheck` set in tests.
+    if (!_appCheck) {
+      _appCheck = [[GIDAppCheck alloc] initWithAppCheckTokenFetcher:nil];
+    }
+    [_appCheck prepareForAppCheckWithCompletion:^(FIRAppCheckToken * _Nullable token,
+                                                  NSError * _Nullable error) {
+      if (token) {
+        self->_useAppCheckToken = YES;
```

Comment:
I think `self->` still is needed to suppress the warning below. Otherwise, we could do the weakify/strongify dance, but I was trying to match the style in the file.

![image](https://github.com/google/GoogleSignIn-iOS/assets/4029922/3682b073-9f9d-41ac-8eb1-7ec5dd9f48ea)